src/components/common/Directors.vue

```diff
@@ -1124,45 +1128,53 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
         director.deliveryAddress = this.inProgressAddress
       }
 
-      if (!Object.values(this.inProgressMailAddress).every(el => el === undefined)) {
-        director.mailingAddress = this.inProgressMailAddress
-      }
+      if (this.isBComp) {
+        if (!Object.values(this.inProgressMailAddress).every(el => el === undefined)) {
+          director.mailingAddress = this.inProgressMailAddress
+        }
 
-      if (this.inheritDeliveryAddress) {
-        director.mailingAddress = director.deliveryAddress
+        if (this.inheritDeliveryAddress) {
+          director.mailingAddress = director.deliveryAddress
+        }
       }
 
       /* COMPARE changes to original director data, for existing directors */
       if (director.actions.indexOf(Actions.APPOINTED) < 0) {
-        const origDirector = this.directorsOriginal.filter(el => el.id === id)[0]
```

Comment:
"filter" goes through the whole list, while "find" stops when it finds a match.

Also the new code checks that director was found before dereferencing ("[0]").